cpp/meson.options

```diff
@@ -66,6 +66,22 @@ option(
     type: 'string',
     description: 'Arbitrary string that identifies the kind of package (for informational purposes)',
 )
+option('parquet', type: 'feature', description: 'Build the Parquet libraries')
```

Comment:
There are a few differences with the CMake configuration worth calling out.

First, CMake has an extra option for `PARQUET_MINIMAL_DEPENDENCY`. This seems to maybe not be used all of that often, so I did not bother implementing here. Let me know if that is a mistake.

CMake also requires that when either the executables or examples are built that you use static libraries. My assumption is that was done to make the CMake configuration easier, so I did not place that same restriction in the Meson configuration.

Finally, CMake will fail if a pre-installed version of OpenSSL is not found and encryption is required. This configuration is not as strict, and will use Meson's wrap system to resolve the OpenSSL dependency if it is not provided by the system